packages/fx-core/src/component/middleware/processEnvRestorerMW.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { Middleware, NextFunction } from "@feathersjs/hooks";
+import { CoreHookContext } from "../../core/types";
+import _ from "lodash";
+
+export const ProcessEnvRestorerMW: Middleware = async (
```

Comment:
can it be part of Huajie's envLoaderMW? so we don't need to add another middleware?

Reply:
Adding it to existing middleware will add a big `try finally` block in existing middleware code. which is kind of messy and harder test. But the advantage is we don't need to modify the decorator list for each lifecycle.

Reply:
One thing to consider is whether `envLoaderMW` is responsible to restore env at the end. My answer is yes (I'm not aware of a case that don't need to restore the env). Also, if we make 2 separate middleware, we need a way to ensure everyone (including new members) knows he/she needs to use `ProcessEnvRestorerMW` along with `envLoaderMW`. Failing to do so may introduce bugs. You can discuss with Huajie to make the final decision.

By the way, can you share more about your concern of `messy and harder test`?

Reply:
"By the way, can you share more about your concern of messy and harder test?" There are no test for envloaderMW, I'll have to write from scratch.